pytorch_lightning/callbacks/progress/tqdm_progress.py

```diff
@@ -216,87 +225,74 @@ def on_sanity_check_end(self, trainer, pl_module):
             self.main_progress_bar.set_postfix(self.get_metrics(trainer, pl_module))
 
     def on_train_end(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> None:
         self.main_progress_bar.close()
 
     def on_validation_start(self, trainer, pl_module):
-        super().on_validation_start(trainer, pl_module)
         if trainer.sanity_checking:
-            reset(self.val_progress_bar, total=sum(trainer.num_sanity_val_batches), current=self.val_batch_idx)
+            self.val_progress_bar.total = sum(trainer.num_sanity_val_batches)
         else:
-            if trainer.state.fn == pl.trainer.states.TrainerFn.FITTING:
-                self._update_bar(self.main_progress_bar)  # fill up remaining
             self.val_progress_bar = self.init_validation_tqdm()
-            reset(self.val_progress_bar, total=self.total_val_batches, current=self.val_batch_idx)
+            self.val_progress_bar.total = convert_inf(self.total_val_batches)
 
     def on_validation_batch_end(self, trainer, pl_module, outputs, batch, batch_idx, dataloader_idx):
-        super().on_validation_batch_end(trainer, pl_module, outputs, batch, batch_idx, dataloader_idx)
         if self._should_update(self.val_batch_idx):
-            self._update_bar(self.val_progress_bar)
+            _update_n(self.val_progress_bar, self.val_batch_idx)
             if trainer.state.fn == pl.trainer.states.TrainerFn.FITTING:
-                self._update_bar(self.main_progress_bar)
+                _update_n(self.main_progress_bar, self.train_batch_idx + self._val_processed)
 
     def on_validation_epoch_end(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> None:
-        if self.is_enabled:
-            self._update_bar(self.val_progress_bar)
+        _update_n(self.val_progress_bar, self._val_processed)
 
     def on_validation_end(self, trainer, pl_module):
         if self.main_progress_bar is not None and trainer.state.fn == pl.trainer.states.TrainerFn.FITTING:
             self.main_progress_bar.set_postfix(self.get_metrics(trainer, pl_module))
         self.val_progress_bar.close()
 
     def on_test_start(self, trainer, pl_module):
-        super().on_test_start(trainer, pl_module)
         self.test_progress_bar = self.init_test_tqdm()
         self.test_progress_bar.total = convert_inf(self.total_test_batches)
 
     def on_test_batch_end(self, trainer, pl_module, outputs, batch, batch_idx, dataloader_idx):
-        super().on_test_batch_end(trainer, pl_module, outputs, batch, batch_idx, dataloader_idx)
         if self._should_update(self.test_batch_idx):
-            self._update_bar(self.test_progress_bar)
+            _update_n(self.test_progress_bar, self.test_batch_idx)
 
     def on_test_epoch_end(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> None:
-        if self.is_enabled:
-            self._update_bar(self.test_progress_bar)
+        _update_n(self.test_progress_bar, self.test_batch_idx)
```

Comment:
this will update the bar when it is disabled?

Reply:
`_update_n` checks for disabled